infra/src/main/java/com/redhat/service/smartevents/infra/exceptions/definitions/platform/ProvisioningFailureException.java

```diff
@@ -0,0 +1,14 @@
+package com.redhat.service.smartevents.infra.exceptions.definitions.platform;
+
+import com.redhat.service.smartevents.infra.exceptions.ProvidesReason;
+
+public class ProvisioningFailureException extends InternalPlatformException implements ProvidesReason {
```

Comment:
I only have one exception type for where we explicitly handle certain scenarios... I could replace with more explicit types for each scenario and delete the `ProvidesReason` interface.. but open for opinions.